debugger: Use DAP schema to configure daps

Anthony-Eid opened the pull request:
This PR allows DAPs to define their own schema so users can see completion items when editing their debug.json files.

Users facing this aren’t the biggest chance, but behind the scenes, this affected a lot of code because we manually translated common fields from Zed's config format to be adapter-specific. Now we store the raw JSON from a user's configuration file and just send that. 

I'm ignoring the Protobuf CICD error because the DebugTaskDefinition message is not yet user facing and we need to deprecate some fields in it. 

Release Notes:

- debugger beta: Show completion items when editing debug.json
- debugger beta: Breaking change, debug.json schema now relays on what DAP you have selected instead of always having the same based values.

Anthony-Eid committed:
Initial work to get completions based on a dap's schema

Anthony-Eid committed:
Stuff

Anthony-Eid committed:
More WIP

Co-authored-by: Remco Smits <djsmits12@gmail.com>

Anthony-Eid committed:
more wip

Anthony-Eid committed:
more wip

Anthony-Eid committed:
more wip

Anthony-Eid committed:
Get locators somewhat working again

Anthony-Eid committed:
more wip

Co-authored-by: Cole Miller <m@cole-miller.net>
Co-authored-by: Remco Smits <djsmits12@gmail.com>

cole-miller committed:
WIP

Co-authored-by: Anthony Eid <hello@anthonyeid.me>

Anthony-Eid committed:
More wip yayyyyy

Co-authored-by: Cole Miller <m@cole-miller.net>

osiewicz commented on crates/dap_adapters/src/codelldb.rs:
```diff
@@ -109,10 +95,333 @@ impl DebugAdapter for CodeLldbDebugAdapter {
+    }
+
+    fn dap_schema(&self) -> serde_json::Value {
+        json!({
```
Can we store these in a separate `.json` file that gets `include_str`'d in here?

Anthony-Eid replied:
Yes! There's a small chance that I might do that in another PR so we don't block letting beta users in

Anthony-Eid committed:
More wip

Anthony-Eid committed:
More wip

cole-miller committed:
WIP

Co-authored-by: Anthony Eid <hello@anthonyeid.me>

cole-miller committed:
Clean up

cole-miller committed:
Remove the name property in a couple of places

cole-miller committed:
remove mentions of external terminal

cole-miller committed:
replace mentions of workspaceFolder

cole-miller committed:
WIP

Anthony-Eid committed:
Implment dap schema in extension API

Anthony-Eid committed:
Fix project search panic

The panic occured when querying a second search in the project search
multibuffer while there were dirty buffers.

Anthony-Eid committed:
Polish

Anthony-Eid committed:
merge with main

Anthony-Eid committed:
cicd errors

Anthony-Eid committed:
merge with main

Anthony-Eid committed:
Remove wasm cache test

This test will cause problems for future extension changes so it's easier to remove

Anthony-Eid committed:
Fix clippy errors

Anthony-Eid committed:
Cargo fmt

ConradIrwin commented on crates/dap_adapters/src/ruby.rs:
```diff
@@ -66,34 +248,25 @@ impl DebugAdapter for RubyDebugAdapter {
             format!("--port={}", port),
             format!("--host={}", host),
         ];
-        if delegate.which(launch.program.as_ref()).await.is_some() {
```
Sorry for the delay, fix here: https://github.com/zed-industries/zed/pull/32407

nudded replied:
amazing thanks!
